### PR TITLE
feat: add ability to build raids via Maas-api

### PR DESCRIPTION
**Pull request by dpoluyanov**
Yet this fixes problem with deserialization machine-objects with declared RAID structure

### CODE REVIEW

File: api/machines.go
Comment (cpg1111) [inline api/machines.go:11]: `accept_all` is intended to add machines into MAAS, possibly by a non-admin user, and then later have an admin user approve these machines. Why pull this in for RAID?

**Review by cpg1111 — CHANGES_REQUESTED** [review]
Thanks for the contribution! The RAID changes are looking good, however, there is a question around the use of `accept_all`, see inline comment.

File: api/machines.go
Comment (dpoluyanov) [reply api/machines.go:11]: We,ve use it in automation scenarios, without admin or user intervention, from host discovery to setup custom image and push host to production, so accept_all was more fast approach in comparison with iteration through list of machines and acceptance only New one.
Probably I could separate it to separate PR if it looks like unsuitable api call for gomaasclient

Comment (cpg1111) [reply api/machines.go:11]: Yeah, I think it's a fine addition, just ideal to have it as a separate PR since it is not RAID related.

Comment (dpoluyanov) [reply api/machines.go]: Removed `accept_all` from this branch
Will create separate PR later

**Comment by SK1Y101** [comment]
Hey @dpoluyanov the changes look okay to me, and I see you've addressed the feedback from @cpg1111.
I'm not super familiar with Go, so I'm going to have to call in a second opinion before we merge though.

**Comment by donwlewis** [comment]
Any chance this will get merged in soon?

**Review by r00ta — CHANGES_REQUESTED** [review]
Hi @donwlewis , could you add a test for this feature? You just need to put a `raid.json` under `test/testdata/maas` with a response from MAAS to be deserialized by this code

**Pull request merged**
